### PR TITLE
[backport] ndimage: coordinate rounding fix for order=0 interpolation

### PR DESCRIPTION
**Pull request by emcastillo**
Backport of #4552

### CODE REVIEW

**Comment by emcastillo** [comment]
[automatic post] Jenkins, test this please.

File: cupyx/scipy/ndimage/_interp_kernels.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def _generate_interp_custom(coord_func, ndim, large_int, yshape, mode, cval,
         for j in range(ndim):
             # determine nearest neighbor
             ops.append("""
-            {int_t} cf_{j} = ({int_t})lrint((double)c_{j});
+            {int_t} cf_{j} = ({int_t})floor((double)c_{j} + 0.5);''')
```
Comment (toslunar) [inline cupyx/scipy/ndimage/_interp_kernels.py:246]:
```suggestion
            {int_t} cf_{j} = ({int_t})floor((double)c_{j} + 0.5);
```

**Comment by asi1024** [comment]
Could you check test failure?

**Comment by emcastillo** [comment]
PTAL :)

**Comment by toslunar** [comment]
pfnCI, test this please.

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**